File: XBOOK/XBOOK.Dapper/ViewModels/BuyInvoiceViewModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace XBOOK.Dapper.ViewModels
{
    public class BuyInvoiceViewModel
    {
        public int InvoiceId { get; set; }
        public string InvoiceNumber { get; set; }
        public string InvoiceSerial { get; set; }
        public string SupplierName { get; set; }
        public string Note { get; set; }
        public DateTime IssueDate { get; set; }
        public DateTime DueDate { get; set; }
        public decimal Amount { get; set; }
        public decimal AmountPaid { get; set; }
        public int SupplierID { get; set; }
        public string ContactName { get; set; }
        public string BankAccount { get; set; }
    }
}

File: XBOOK/XBOOK.Web/ClientApp/src/app/pages/_shared/models/invoice/buy-invoice-view.model.ts
import { SaleInvoiceDetail } from './sale-invoice-detail.model';
import { ClientView } from '../client/client-view.model';
import { PaymentView } from './payment-view.model';

export class BuyInvoiceView {
  invoiceId: number;
  invoiceSerial: string;
  invoiceNumber: string;
  issueDate: Date;
  dueDate: Date;
  supplierID: number;
  supplierName: string;
  address: string;
  taxCode: string;
  contactName: string;
  email: string;
  reference: string;
  subTotal: number;
  discRate: number;
  discount: number;
  vatTax: number;
  amountPaid: number;
  note: string;
  term: string;
  status: string;
  bankAccount: string;
  saleInvDetailView: SaleInvoiceDetail[];
  paymentView: PaymentView[];
  clientData: ClientView[];
}
